packages/core/src/base/Time.ts

```diff
@@ -19,68 +38,78 @@ export class Time {
   }
 
   /**
-   * Constructor of the Time.
+   * The delta time in seconds from the last frame to the current frame.
    */
-  constructor() {
-    this._clock = performance ? performance : Date;
-
-    this._timeScale = 1.0;
-    this._deltaTime = 0.0001;
-
-    const now = this._clock.now();
-    this._startTime = now;
-    this._lastTickTime = now;
+  get deltaTime(): number {
+    return this._deltaTime;
   }
 
-  reset() {
-    this._lastTickTime = this._clock.now();
+  /**
+   * The unscaled delta time in seconds from the last frame to the current frame.
+   */
+  get unscaledDeltaTime(): number {
+    return this._unscaledDeltaTime;
   }
 
   /**
-   * Current Time
+   * The amount of elapsed time in seconds since the start of the engine.
    */
-  get nowTime(): number {
-    return this._clock.now();
+  get elapsedTime(): number {
+    return this._elapsedTime;
   }
 
   /**
-   * Time between two ticks
+   * The amount of unscaled elapsed time in seconds since the start of the engine.
    */
-  get deltaTime(): number {
-    return this._deltaTime;
+  get unscaledElapsedTime(): number {
+    return this._unscaledElapsedTime;
   }
 
   /**
-   * Scaled delta time.
+   * Constructor of the Time.
    */
-  get timeScale(): number {
-    return this._timeScale;
-  }
-  set timeScale(s) {
-    this._timeScale = s;
+  constructor() {
+    this._lastSystemTime = performance.now() / 1000;
   }
 
   /**
-   * Unscaled delta time.
+   * @internal
    */
-  get unscaledDeltaTime(): number {
-    return this._deltaTime / this._timeScale;
+  _reset() {
+    this._lastSystemTime = performance.now() / 1000;
   }
 
   /**
-   * The elapsed time, after the clock is initialized.
+   * @internal
    */
-  get timeSinceStartup(): number {
-    return this.nowTime - this._startTime;
+  _update(): void {
+    const currentSystemTime = performance.now() / 1000;
+
+    const unscaledDeltaTime = currentSystemTime - this._lastSystemTime;
+    this._unscaledDeltaTime = unscaledDeltaTime;
+    this._unscaledElapsedTime += unscaledDeltaTime;
+
+    const deltaTime = Math.min(unscaledDeltaTime, this.maximumDeltaTime) * this.timeScale;
+    this._deltaTime = deltaTime;
+    this._elapsedTime += deltaTime;
+    this._frameCount++;
+
+    this._lastSystemTime = currentSystemTime;
   }
 
   /**
-   * Call every frame, update delta time and other data.
+   * @internal
    */
-  public tick(): void {
-    const now = this.nowTime;
-    this._deltaTime = (now - this._lastTickTime) * this._timeScale;
-    this._lastTickTime = now;
-    this._frameCount++;
+  _updateSceneShaderData(shaderData: ShaderData): void {
+    const elapsedTimeValue = this._elapsedTimeValue;
+    const deltaTimeValue = this._deltaTimeValue;
```

Comment:
`const {_elapsedTimeValue, _deltaTimeValue} = this;`

Reply:
done